packit/api.py

```diff
@@ -140,6 +140,7 @@ def update_dist_git(
         commit_title: str,
         commit_msg: str,
         sync_default_files: bool = True,
+        pkg_tool: str = "",
```

Comment:
no, we want to pass this all the way down into `upload_to_lookaside_cache()` where we want to use the `pkg_tool` value from the config (which defaults to `fedpkg`) if this argument is not set

Reply:
I see. To me this seems a little bit too late (or too deep in the stack) for this pkg_tool-or-config decision. Would having it in `cli/update_dist_git.py` make sense? I mean override `config.pkg_tool` with the value provided by `--pkg-tool`. Than later code could use just that.

Reply:
Too late/deep for what?

The path is `cli.update_dist_git` -> `API.update_dist_git` -> `API._handle_sources` -> `DistGit.upload_to_lookaside_cache` where we first need the value/decision in the `upload_to_lookaside_cache`.

If the user does not specify `--pkg-tool` we don't even run `API._handle_sources` because `add_new_sources` is `False` so it makes no sense do do the pkg_tool-or-config decision sooner.
And if the user specifies it, it's decided and we already pass the value all the way down.

Yes, I could make the decision ASAP and pass it all the way, but why if the code along the way doesn't need it?
To me it's like defining a variable/value at start of a method when I first use it somewhere in the middle.

Reply:
Yeah, maybe you are right.